### ISSUE TITLE
Some bibliography cross-references don't render in number format

### ISSUE THREAD

**Issue by manuel489** [metanorma/metanorma-iso#469]
*Found while working in https://github.com/metanorma/iso-690/issues/12*

In [iso-690](https://github.com/metanorma/iso-690) document, all bibliography cross-references are expected to render in number format.
However, the auto-fetched ones are rendering different:

DOC output:
![issue4-doc](https://user-images.githubusercontent.com/33627611/107148500-cf741900-6929-11eb-8b83-bbceee43c636.PNG)

Original source:
![issue4-pdf](https://user-images.githubusercontent.com/33627611/107148501-d1d67300-6929-11eb-9aa1-7627adab30e5.PNG)

**Comment by opoudjis** [comment]
The behaviour you are encountering is what the Rice model document does (https://www.iso.org/files/live/sites/isoorg/files/developing_standards/docs/en/model_document-rice_model.pdf: e..g not `[1]`, but `ISO 3696`); that's why we've implemented it like that. There is no explicit discussion of citations in ISO/IEC DIR 2, but there is a general statement that standards must be referenced by standard number; if anything, the example of a citation of a  non-standard document is Author-Date (https://www.iso.org/sites/directives/current/part2/index.xhtml#_idTextAnchor114, Example 2).

In other words, the behaviour of Metanorma, overriding numbered references, appears to be the behaviour expected by ISO, and I would rather not change it to comply with a manually compiled document, nor implement an override, without explicit advice from ISO editors.

**Comment by ronaldtse** [comment]
> I would rather not change it to comply with a manually compiled document

ISO 690 was originally encoded in Metanorma... but updated in Word.

The issue is about reference style -- in ISO we should just follow the new [ISO house style](https://www.iso.org/ISO-house-style.html) which states the following.

--------

### CROSS-REFERENCES TO A NUMBERED DOCUMENT

Always use the standard identifier (document number and type) rather than the title when referring to an ISO or IEC document in the text, including when a specific element (clause, table, figure, etc.) is referred to.

Only write “International Standard”, “Technical Specification”, etc. when making a general reference. For example, “X is specified in other International Standards, e.g. ISO 12345”. Do not write “ISO Standard”, “ISO specification” or other alternatives.

Only write out the title of a numbered document in Clause 2 and in the Bibliography.

When referring to the whole document, use an undated document number unless it is necessary that the user refers to a specific edition:

“…in accordance with ISO 12345.”
Use the order of “document number:date, element” when referring to a specific element:

“…in accordance with ISO 12345:2018, 4.3.9.”
“…in accordance with ISO/IEC TR 6789:2020, Table 3.”
“…in accordance with ISO 12345:2018/Amd.1:2019, Figure C.1.”
When referring to a document that is not published by ISO or IEC, use the same style as for an ISO or IEC document if a document number is available:

“…in accordance with IETF RFC 2046.”
“…in accordance with Rec. ITU-T X.96:2000, Annex F.”


### CROSS-REFERENCES TO AN UNNUMBERED DOCUMENT
To refer to unnumbered documents in the Bibliography (e.g. a journal or literature reference), use any of the following styles:

“…see Reference [6]”
“…an initial interlaboratory test[6] was carried out…”
“…in the initial interlaboratory test[6]”
“…in the six subsequent interlaboratory tests[7]–[12]”
For non-ISO or IEC documents, it is possible to provide both a document number and a superscript callout:

“…as described in IEEE 802.3[9]”
If it is considered necessary to write the title, give the correct wording consistently throughout the text, not a shortened name or abbreviation:

“…for example, in the United Nations’ Recommendations on the Transport of Dangerous Goods – Model Regulations[1]”
It is permitted to use a mixture of these styles within the text when referring to bibliographical entries.

To refer to unnumbered documents in Clause 2, use the full title of the document without a superscript callout.

---------

AND

---------

###ORDER OF BIBLIOGRAPHIC REFERENCES

The Bibliography is always numbered. There are two ways of structuring bibliographical entries:

By numerical order, followed by literature references:
- ISO documents followed by IEC documents and then other international standards;
- regional standards;
- national standards;
- literature references.

In the order in which the references are cited in the text.

For any type of standard that is not published by ISO or IEC, it is possible to provide an online reference to the bibliographical entry in a footnote:

> [7] Standard No I.C.C. 167[1], Determination of the protein content in cereal and cereal products for food and animal feeding stuffs according to the Dumas combustion method
>   [1] Available at http://www.icc.or.at.

The presentation of references follows the guidelines in ISO 690.

For a literature reference, use the format of: Author/organization name. Document title. Location: publisher, date

LEBOFFE, M.J. and PIERCE, B.E. Microbiology: laboratory theory and application. Englewood, Colorado: Morton Publishing Company, 2010

ADVANCED LIFE SUPPORT GROUP. Acute medical emergencies: the practical approach. London: BMJ Books, 2001
For a journal reference, use the format of: Author. Article title. Journal title. Date, edition number, page number. Online reference

CROPLEY, A. J. Creativity and mental health in everyday life. Creativity Research Journal. 1990, 3(3), 167–178. doi:10.1080/10400419009534351
For an online reference, use the format of: Author. Article title. Publication title. Location, date [viewed date]. URL

NUNN, G. Don’t ditch the adverb, the emoji of writing. The Guardian [online]. London, UK, 2 April 2019 [viewed 27 April 2020]. Available from: https://www.theguardian.com/commentisfree/2019/apr/29/dont-ditch-the-adverb-the-emoji-of-writing

**Comment by opoudjis** [comment]
@ronaldtse, the house style is confirming the approach I have taken: standards  are *always* referred to by document identifier, whether in bibliography or normative references, and numbers are only used as citations for "unnumbered' documents (by which they mean documents without a document identifier), "e.g. a journal or literature reference". 

It does now explicitly confirm that "unnumbered" normative references are not referred to by document number, which we are now banning.

Numbers are optional for non-ISO/IEC standards documents, in addition to the document identifier ("it is possible to"); we're not doing that, we're just giving the document identifier.

Metanorma is complying to these guidelines, which are just clarifications of ISO/IEC DIR 2 and which were already exemplified in the Rice document. Nothing further to do here.

**Comment by ronaldtse** [comment]
In any case, here "FIPS 202" should be "NIST FIPS 202", "RFC 3986" should be "IETF RFC 3986"?

**Comment by opoudjis** [comment]
The docidentifiers for NIST retrieved from relaton are inconsistent: NIST FIPS 81 has the docidentifier `<docidentifier type="NIST">NIST FIPS 81</docidentifier>`, while NIST FIPS 186 has the docidentifier `<docidentifier type="NIST">FIPS 86</docidentifier>`. But the identifier given in the references is correct (I insert the prefix if not already present), and that given to the eref should match that. Fixing.

**Comment by opoudjis** [comment]
Prefixes were only being added to ISO/IEC document identifiers; that's a bug. Removing that constraint. Will prevent prefixes being added if they are already present in the docidentifier.
